### PR TITLE
fix(vue): dataField optional for DataSearch for rs-api

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
## Pre-requisite
Read this https://stackoverflow.com/questions/56904327/access-props-value-in-another-props-validator

## What does this PR do?
- fix validation for dataField when `enableAppbse=true`

## Test
https://www.loom.com/share/4f6b6af7973441dfb53c758ce8d02623

## Docs
https://github.com/appbaseio/Docs/pull/141

### CODE REVIEW

File: packages/vue/src/components/search/DataSearch.jsx [hunk 1]
```diff
@@ -262,15 +264,31 @@ const DataSearch = {
 		},
 	},
 	methods: {
+		validateDataField() {
+			const propName = 'dataField';
+			const componentName = 'DataSearch';
```
Comment (bietkul) [inline packages/vue/src/components/search/DataSearch.jsx]: @ShahAnuj2610 Use it from `name` property.
https://github.com/appbaseio/reactivesearch/blob/bb371f05df1dce890f0b1d6bac6a6288922defae/packages/vue/src/components/search/DataSearch.jsx#L54

File: packages/vue/src/components/search/DataSearch.jsx [hunk 2]
```diff
@@ -262,15 +264,31 @@ const DataSearch = {
...
 		updateQueryOptions() {
-			if (
-				this.customHighlight
-				&& typeof this.customHighlight === 'function'
-			) {
-				this.setCustomHighlightOptions(
-					this.componentId,
-					this.customHighlight(this.$props),
-				);
+			if (this.customHighlight && typeof this.customHighlight === 'function') {
```
Comment (bietkul) [inline packages/vue/src/components/search/DataSearch.jsx:304]: @ShahAnuj2610 Why formatting changes?

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/vue/src/components/search/DataSearch.jsx (hunk 2 above)
Comment (ShahAnuj2610) [reply packages/vue/src/components/search/DataSearch.jsx:304]: not sure. I'm using prettier and eslint as per repo config.

File: packages/vue/src/components/search/DataSearch.jsx (hunk 1 above)
Comment (ShahAnuj2610) [reply packages/vue/src/components/search/DataSearch.jsx]: done

Comment (bietkul) [reply packages/vue/src/components/search/DataSearch.jsx]: @ShahAnuj2610 How is it done?
You're still hardcoding the `componentName`, it should be `this.name`.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/vue/src/components/search/DataSearch.jsx (hunk 1 above)
Comment (ShahAnuj2610) [reply packages/vue/src/components/search/DataSearch.jsx]: @bietkul check this https://github.com/appbaseio/reactivesearch/pull/1448/files#diff-29c2691a20b54ace57eca812d2d9ca86R269

**Review by bietkul — APPROVED** [review]

**Pull request merged**